### PR TITLE
Remove P3A protobuf implementation

### PR DESCRIPTION
**Pull request by rillian**
Disable the remaining P2A pathway for reporting protobuf-format P3A messages and remove the associated code. This completes the transition to json-format reporting.

Resolves https://github.com/brave/brave-browser/issues/23146

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/p3a/p3a_message.cc [hunk 1]
```diff
@@ -19,53 +19,6 @@ namespace brave {
 MessageMetainfo::MessageMetainfo() = default;
 MessageMetainfo::~MessageMetainfo() = default;
 
-void GenerateP3AMessage(uint64_t metric_hash,
-                        uint64_t metric_value,
-                        const MessageMetainfo& meta,
-                        brave_pyxis::RawP3AValue* p3a_message) {
```
Comment (iefremov) [inline components/p3a/p3a_message.cc:25]: can we remove the entire `components/brave_prochlo` now?

Comment (iefremov) [reply components/p3a/p3a_message.cc:25]: also pls drop other unused includes like trace_event.h or sha2.h

Comment (rillian) [reply components/p3a/p3a_message.cc:25]: Yes! The stats aggregator has its own copy, so this should be safe to remove.

Comment (rillian) [reply components/p3a/p3a_message.cc:25]: Removed in 11688513e4fc8728509ddf8d99005c84ce34a531.

File: components/p3a/brave_p3a_service.cc [hunk 2]
```diff
@@ -187,27 +180,21 @@ void BraveP3AService::Init(
...
-    base::StringPiece histogram_name,
-    uint64_t value) {
+std::string BraveP3AService::Serialize(base::StringPiece histogram_name,
+                                       uint64_t value) {
   // TRACE_EVENT0("brave_p3a", "SerializeMessage");
   // TODO(iefremov): Maybe we should store it in logs and pass here?
   // We cannot directly query |base::StatisticsRecorder::FindHistogram| because
   // the serialized value can be obtained from persisted log storage at the
   // point when the actual histogram is not ready yet.
-  const uint64_t histogram_name_hash = base::HashMetricName(histogram_name);
```
Comment (iefremov) [inline components/p3a/brave_p3a_service.cc:198]: perhaps we can drop metrics_hashes.h

File: components/p3a/BUILD.gn [hunk 3]
```diff
@@ -33,7 +31,6 @@ static_library("p3a") {
 
   deps = [
     "//base",
-    "//brave/components/brave_prochlo:prochlo_proto",
```
Comment (iefremov) [inline components/p3a/BUILD.gn:36]: i think we can also drop protobuf:protobuf_lite` and `//services/network/public/mojom`

File: components/p3a/brave_p3a_uploader.cc [hunk 4]
```diff
@@ -7,7 +7,6 @@
 
 #include <utility>
 
-#include "base/base64.h"
 #include "net/base/load_flags.h"
 #include "services/network/public/cpp/resource_request.h"
 #include "services/network/public/cpp/shared_url_loader_factory.h"
```
Comment (iefremov) [inline components/p3a/brave_p3a_uploader.cc:12]: `url_response_head` seems not used, together with a dep in BUILD.gn

**Review by iefremov — APPROVED** [review]

File: components/p3a/brave_p3a_uploader.cc (hunk 4 above)
Comment (rillian) [reply components/p3a/brave_p3a_uploader.cc:12]: This is used to access the return code in `BraveP3AUploader::OnUploadComplete`.

`BraveP3AService::OnLogUploadComplete` checks for a 200 before retiring the log, so I think we still need this.

The callback also returns an error flag which comes directly from `SimpleUrlLoader::NetError()` so we could use that instead, but I think checking for `200` is better.

Comment (rillian) [reply components/p3a/brave_p3a_uploader.cc:12]: It looks like `net::Error` can include a check for non-200 response codes. I've filed https://github.com/brave/brave-browser/issues/24339 to address this in a follow-up.

**Comment by rillian** [comment]
squashed review fixes, will land if ci stays green.

**Pull request merged**